Libraries/pbxbuild/Headers/pbxbuild/Tool/Input.h

```diff
@@ -29,51 +29,63 @@ class Environment;
  */
 class Input {
 private:
-    pbxproj::PBX::BuildFile::shared_ptr       _buildFile;
-    Target::BuildRules::BuildRule::shared_ptr _buildRule;
+    std::string                               _path;
     pbxspec::PBX::FileType::shared_ptr        _fileType;
 
 private:
-    std::string                               _path;
-    ext::optional<std::string>                _localization;
+    Target::BuildRules::BuildRule::shared_ptr _buildRule;
     ext::optional<std::string>                _fileNameDisambiguator;
 
+private:
+    ext::optional<std::string>                _localization;
+    ext::optional<std::string>                _localizationGroupIdentifier;
+
+private:
+    ext::optional<std::vector<std::string>>   _attributes;
+    ext::optional<std::vector<std::string>>   _compilerFlags;
+
 public:
     Input(
-        pbxproj::PBX::BuildFile::shared_ptr const &buildFile,
-        Target::BuildRules::BuildRule::shared_ptr const &buildRule,
-        pbxspec::PBX::FileType::shared_ptr const &fileType,
         std::string const &path,
+        pbxspec::PBX::FileType::shared_ptr const &fileType,
+        Target::BuildRules::BuildRule::shared_ptr const &buildRule,
+        ext::optional<std::string> const &fileNameDisambiguator,
         ext::optional<std::string> const &localization,
-        ext::optional<std::string> const &fileNameDisambiguator);
+        ext::optional<std::string> const &localizationGroupIdentifier,
+        ext::optional<std::vector<std::string>> const &attributes,
+        ext::optional<std::vector<std::string>> const &compilerFlags);
+    Input(std::string const &path, pbxspec::PBX::FileType::shared_ptr const &fileType);
     ~Input();
 
 public:
     /*
-     * The build file entry saying *to* build this file. May be nullptr.
-     */
-    pbxproj::PBX::BuildFile::shared_ptr const &buildFile() const
-    { return _buildFile; }
-
-    /*
-     * The build file entry saying *how* to build this file.
+     * The resolved absolute path to the file.
      */
-    Target::BuildRules::BuildRule::shared_ptr const &buildRule() const
-    { return _buildRule; }
+    std::string const &path() const
+    { return _path; }
 
     /*
-     * The type of this file.
+     * The type of this file. Optional.
      */
     pbxspec::PBX::FileType::shared_ptr const &fileType() const
     { return _fileType; }
 
 public:
     /*
-     * The resolved absolute path to the file.
+     * The build rule saying *how* to build this file. Optional.
      */
-    std::string const &path() const
-    { return _path; }
+    Target::BuildRules::BuildRule::shared_ptr const &buildRule() const
+    { return _buildRule; }
+
+    /*
+     * A disambiguation identifier for files that have multiple with the
```

Comment:
"that have multiple with the same base name"

Confusing.  You probably ought to reword this to something like "A disambiguation identifier for different files with the same base name..."